src/main/java/com/milamber_brass/brass_armory/entities/BAArrowEntity.java

```diff
@@ -249,11 +256,12 @@ public void tick() {
                 this.remove();
             }
         }
-        // check that the arrow is not in the ground and has been flying for half a second.
-        else if (this.isArrowType(ArrowType.LASER) && !this.inGround && this.flightTime > 10) {
-            // make the arrows Y position only decrease by 0.05 every tick (1 block per second).
-            this.moveTo(this.getX(), this.yo - .0005, this.getZ());
-            // if the arrow enters an unloaded chunk, remove it.
+        // check that the arrow is not in the ground and has been flying for 1/5 a second.
+        else if (this.isArrowType(ArrowType.LASER) && !this.inGround && this.flightTime > 4) {
+            // call super.tick() twice to speed up arrow movement.
+            super.tick();
+            super.tick();
```

Comment:
Doing so, at least in my testing,really messed with the block/entity hit detection. 

It might be better to remove the third call to super.tick(), but basically calling super.tick() on the arrow just runs all of the positional code a second time, thereby moving the arrow faster, as well as the hit detection code.

Reply:
That does not sound like a good solution, doing that really can mess up a lot cause some parts of the onUpdate could should only be called once per tick (well, the rest of the game and other mods  calculate with it only being called once per tick).

If you just want to change the speed, adjust the values in the constructor, iirc projectiles have a special constructor where the direction and velocity gets set.

Reply:
I'll look into that.
 for now, I'm going to merge this branch and look into that in a separate branch from the melee weapons